internal/cli/kraft/compose/create/create.go

```diff
@@ -0,0 +1,409 @@
+	mplatform "kraftkit.sh/machine/platform"
+)
+
+type CreateOptions struct {
+	Composefile string `noattribute:"true"`
+}
+
+func NewCmd() *cobra.Command {
+	cmd, err := cmdfactory.New(&CreateOptions{}, cobra.Command{
+		Short:   "Create a compose project",
+		Use:     "create [FLAGS]",
+		Args:    cobra.NoArgs,
+		Aliases: []string{},
+		Long:    "Create the services and networks for a project.",
+		Example: heredoc.Doc(`
+			# Create the networks and services without running them
+			$ kraft compose create 
+		`),
+		Annotations: map[string]string{
+			cmdfactory.AnnotationHelpGroup: "compose",
+		},
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	return cmd
+}
+
+func (opts *CreateOptions) Pre(cmd *cobra.Command, _ []string) error {
+	ctx, err := packmanager.WithDefaultUmbrellaManagerInContext(cmd.Context())
+	if err != nil {
+		return err
+	}
+
+	cmd.SetContext(ctx)
+
+	if cmd.Flag("file").Changed {
+		opts.Composefile = cmd.Flag("file").Value.String()
+	}
+
+	log.G(cmd.Context()).WithField("composefile", opts.Composefile).Debug("using")
+	return nil
+}
+
+func (opts *CreateOptions) Run(ctx context.Context, args []string) error {
+	workdir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	project, err := compose.NewProjectFromComposeFile(ctx, workdir, opts.Composefile)
+	if err != nil {
+		return err
+	}
+
+	if err := project.Validate(ctx); err != nil {
+		return err
+	}
+
+	if err := project.AssignIPs(ctx); err != nil {
+		return err
+	}
+
+	composeController, err := compose.NewComposeProjectV1(ctx)
+	if err != nil {
+		return err
+	}
+
+	embeddedProject, err := composeController.Get(ctx, &composeapi.Compose{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: project.Name,
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	projectNetworks := []metav1.ObjectMeta{}
+	if embeddedProject != nil {
+		projectNetworks = embeddedProject.Status.Networks
+	}
+
+	networkController, err := network.NewNetworkV1alpha1ServiceIterator(ctx)
+	if err != nil {
+		return err
+	}
+
+	networks, err := networkController.List(ctx, &networkapi.NetworkList{})
+	if err != nil {
+		return err
+	}
+
+	// We need to first create the networks with a provided subnet
+	// and then the ones which we will assign IPs to
+	subnetNetworks := []string{}
+	emptyNetworks := []string{}
+	for name, network := range project.Networks {
+		if network.Ipam.Config == nil || len(network.Ipam.Config) == 0 {
+			emptyNetworks = append(emptyNetworks, name)
+		} else {
+			subnetNetworks = append(subnetNetworks, name)
+		}
+	}
+
+	orderedNetworks := append(subnetNetworks, emptyNetworks...)
+
+	for _, networkName := range orderedNetworks {
+		network := project.Networks[networkName]
+		alreadyRunning := false
+		for _, n := range networks.Items {
+			if n.Name == network.Name {
+				alreadyRunning = true
+				break
+			}
+		}
+		if alreadyRunning {
+			continue
+		}
+
+		driver := "bridge"
+		if network.Driver != "" {
+			driver = network.Driver
+		}
```

Comment:
I wonder if we shouldn't expose this as a `const`/`var` like `DefaultNetworkDriver` and then retrieved from the relevant platform code.

Reply:
An alternative could be to look into ``hostSupportedStrategies`` and pick the first available?